Fixed bug in opencascade/boundary_lib.cc

luca-heltai opened the pull request:
Fixes #862

bangerth commented on source/opencascade/boundary_lib.cc:
```diff
@@ -233,13 +233,13 @@ namespace OpenCASCADE
+           tolerance*length, ExcPointNotOnManifold(space_point));
 #else
-    (void)space_point;
+    curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
```
Ah, can we avoid this somehow? I.e., just execute the code always and only test its return value in the Assert. Executing instructions with side effects inside Assert statements is bound to be confusing, as you yourself have just found out.

luca-heltai replied:
We could avoid this, at the price of having a warning at compilation time for release mode, where distance is not used...

bangerth replied:
To make the code easier to read, I'd much rather have something like
```
  result = computation();
  Assert (result == ..., Exc...());
#ifndef DEBUG
  (void)result;
#endif
```
At the very least, it is clear that one has to change only one place if the `computation()` call needs to be adjusted later on. The style where you write it twice simply invites a divergence of code.

luca-heltai commented:
I've removed the #ifdef and added a bogus operation to silence warnings on Release mode. Le me know if you want me to switch back to what it was before...

bangerth commented on source/opencascade/boundary_lib.cc:
```diff
@@ -233,14 +233,11 @@ namespace OpenCASCADE
-#endif
+    double dist = curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
+    Assert(dist < tolerance*length, ExcPointNotOnManifold(space_point));
+    dist *= 2; // Silence compiler warning in Release mode.
```
Just change this to `(void)dist;` and make the variable `const` -- this is the standard idiom we use everywhere. Then please merge!
